### ISSUE TITLE
Multiple labels for ChEBI chemicals

### ISSUE THREAD

**Issue by ddooley** [CompositionalDietaryNutritionOntology/cdno#109]
FoodOn imports ChEBI and CDNO terms, and sometimes a chemical gets imported twice.  I noticed with glycerol:

![image](https://user-images.githubusercontent.com/4000582/197822453-03d819c7-4f8a-453b-9da4-ce6c7d120b7b.png)

The middle one is ChEBI's term, the other two originate in CDNO.  Can CDNO drop its hardcoded labels in favour of a ChEBI import file of the same (i.e. an Ontofox or other fetch of CHEBI content).

This came up with attempt to use robot to convert FoodOn to obo format that errors on multiple labels for a single entity.

**Comment by ddooley** [comment]
I think the CHEBI items in question probably are the following ones.  This is from a query at the FoodOn side.  By just relying on CHEBI names, and having any kind of label CDNO wants indicated by an "alternative term" annotation, we would solve the robot convert to obo format generation error.

```
<http://purl.obolibrary.org/obo/CHEBI_76132>	"(+)-catechin 3-O-gallate"@en	"(+)-catechin-3-O-gallate"
<http://purl.obolibrary.org/obo/CHEBI_9577>	"(-)-alpha-thujone"	"(−)-α-thujone"@en
<http://purl.obolibrary.org/obo/CHEBI_4806>	"(-)-epigallocatechin 3-gallate"	"(−)-epigallocatechin 3-gallate"@en
<http://purl.obolibrary.org/obo/CHEBI_67244>	"(-)-lariciresinol"	"(−)-lariciresinol"@en
<http://purl.obolibrary.org/obo/CHEBI_108>	"(-)-phaseolin"	"(−)-phaseolin"@en
<http://purl.obolibrary.org/obo/CHEBI_77366>	"(6Z,9Z,12Z,15Z,18Z,21Z)-tetracosahexaenoic acid"	"6Z,9Z,12Z,15Z,18Z,21Z-tetracosahexaenoic acid"@en
<http://purl.obolibrary.org/obo/CHEBI_132141>	"1,2-naphthoquinones"	"1,2-napthoquinone"@en
<http://purl.obolibrary.org/obo/CHEBI_132142>	"1,4-naphthoquinones"	"1,4-napthoquinone"@en
<http://purl.obolibrary.org/obo/CHEBI_174910>	"4'-O-methyl-(-)-epicatechin"	"4'-O-methyl-(−)-epicatechin"@en
<http://purl.obolibrary.org/obo/CHEBI_140677>	"4-hydroxy-all-trans-beta-carotene"	"4-hydroxy-all-trans-β-carotene"@en
<http://purl.obolibrary.org/obo/CHEBI_1904>	"4-methylumbelliferone beta-D-glucuronide"	"4-methylumbelliferone β-D-glucuronide"@en
<http://purl.obolibrary.org/obo/CHEBI_18394>	"6-O-alpha-D-glucopyranosyl-D-fructofuranose"	"6-O-α-D-glucopyranosyl-D-fructofuranose"@en
<http://purl.obolibrary.org/obo/CHEBI_67188>	"9-cis-beta-carotene"	"9-cis-β-carotene"@en
<http://purl.obolibrary.org/obo/CHEBI_31342>	"Calcium hypochlorite"	"calcium hypochlorite"@en
<http://purl.obolibrary.org/obo/CHEBI_3815>	"Collagen"	"collagen"@en
<http://purl.obolibrary.org/obo/CHEBI_4767>	"Elastin"	"elastin"@en
<http://purl.obolibrary.org/obo/CHEBI_5054>	"Fibrin"	"fibrin"@en
<http://purl.obolibrary.org/obo/CHEBI_141540>	"Myricanone"	"myricanone"@en
<http://purl.obolibrary.org/obo/CHEBI_167695>	"Punicalagin"	"punicalagin"@en
<http://purl.obolibrary.org/obo/CHEBI_28425>	"alpha-carotene"	"α-carotene"@en
<http://purl.obolibrary.org/obo/CHEBI_10223>	"alpha-cryptoxanthin"	"α-cryptoxanthin"@en
<http://purl.obolibrary.org/obo/CHEBI_27432>	"alpha-linolenic acid"	"α-linolenic acid"@en
<http://purl.obolibrary.org/obo/CHEBI_22470>	"alpha-tocopherol"	"α-tocopherol"@en
<http://purl.obolibrary.org/obo/CHEBI_33270>	"alpha-tocotrienol"	"α-tocotrienol"@en
<http://purl.obolibrary.org/obo/CHEBI_83820>	"amino acid derived from non-proteinacious"@en	"non-proteinogenic amino acid"
<http://purl.obolibrary.org/obo/CHEBI_62967>	"amorphous cellulose"	"cellulose"@en
<http://purl.obolibrary.org/obo/CHEBI_30248>	"beta-D-Gal-(1->3)-beta-D-GlcNAc-(1->3)-beta-D-Gal-(1->4)-D-Glc"	"β-D-Gal-(1→3)-β-D-GlcNAc-(1→3)-β-D-Gal-(1→4)-D-Glc"@en
<http://purl.obolibrary.org/obo/CHEBI_22798>	"beta-D-glucoside"	"β-D-glucoside"@en
<http://purl.obolibrary.org/obo/CHEBI_17579>	"beta-carotene"	"β-carotene"@en
<http://purl.obolibrary.org/obo/CHEBI_10362>	"beta-cryptoxanthin"	"β-cryptoxanthin"@en
<http://purl.obolibrary.org/obo/CHEBI_47771>	"beta-tocopherol"	"β-tocopherol"@en
<http://purl.obolibrary.org/obo/CHEBI_33275>	"beta-tocotrienol"	"β-tocotrienol"@en
<http://purl.obolibrary.org/obo/CHEBI_23086>	"chalcone"@en	"chalcones"
<http://purl.obolibrary.org/obo/CHEBI_27475>	"cyanidin 3-O-beta-D-galactoside"	"cyanidin 3-O-β-D-galactoside"@en
<http://purl.obolibrary.org/obo/CHEBI_37664>	"cyanidin 3-O-beta-D-galactoside chloride"	"cyanidin 3-O-β-D-galactoside chloride"@en
<http://purl.obolibrary.org/obo/CHEBI_74811>	"cyanidin 3-O-beta-D-sambubioside"	"cyanidin 3-O-β-D-sambubioside"@en
<http://purl.obolibrary.org/obo/CHEBI_42202>	"daidzein 7-O-beta-D-glucoside"	"daidzein 7-O-β-D-glucoside"@en
<http://purl.obolibrary.org/obo/CHEBI_31463>	"delphinidin 3-O-beta-D-glucoside"	"delphinidin 3-O-β-D-glucoside"@en
<http://purl.obolibrary.org/obo/CHEBI_47772>	"delta-tocopherol"	"δ-tocopherol"@en
<http://purl.obolibrary.org/obo/CHEBI_33276>	"delta-tocotrienol"	"δ-tocotrienol"@en
<http://purl.obolibrary.org/obo/CHEBI_71230>	"dihydrochalcone"@en	"dihydrochalcones"
<http://purl.obolibrary.org/obo/CHEBI_48039>	"dihydroflavanol"@en	"dihydroflavonols"
<http://purl.obolibrary.org/obo/CHEBI_28661>	"gamma-linolenic acid"	"γ-linolenic acid"@en
<http://purl.obolibrary.org/obo/CHEBI_18185>	"gamma-tocopherol"	"γ-tocopherol"@en
<http://purl.obolibrary.org/obo/CHEBI_33277>	"gamma-tocotrienol"	"γ-tocotrienol"@en
<http://purl.obolibrary.org/obo/CHEBI_74224>	"genistein (1-)"@en	"genistein(1-)"
<http://purl.obolibrary.org/obo/CHEBI_27514>	"genistein 7-O-beta-D-glucoside"	"genistein 7-O-β-D-glucoside"@en
<http://purl.obolibrary.org/obo/CHEBI_133348>	"glycitein 7-(6-O-acetyl-beta-D-glucoside)"	"glycitein 7-(6-O-acetyl-β-D-glucoside)"@en
<http://purl.obolibrary.org/obo/CHEBI_18305>	"hydroquinone O-beta-D-glucopyranoside"	"hydroquinone O-β-D-glucopyranoside"@en
<http://purl.obolibrary.org/obo/CHEBI_16382>	"iodide"	"iodine"@en
<http://purl.obolibrary.org/obo/CHEBI_37168>	"iota-carrageenan"	"ι-carrageenan"@en
<http://purl.obolibrary.org/obo/CHEBI_75832>	"iron sulfate"@en	"iron(2+) sulfate (anhydrous)"
<http://purl.obolibrary.org/obo/CHEBI_75832>	"iron sulfate"@en	"iron(2+) sulfate (anhydrous)"@en
<http://purl.obolibrary.org/obo/CHEBI_31742>	"kaempferol 3-O-beta-D-galactoside"	"kaempferol 3-O-β-D-galactoside"@en
<http://purl.obolibrary.org/obo/CHEBI_31743>	"kaempferol 3-O-beta-D-glucosyl-(1->2)-beta-D-glucoside"	"kaempferol 3-O-β-D-glucosyl-(1→2)-β-D-glucosyl-(1→2)-β-D-glucoside"@en
<http://purl.obolibrary.org/obo/CHEBI_10583>	"kappa-carrageenan"	"κ-carrageenan"@en
<http://purl.obolibrary.org/obo/CHEBI_37167>	"lambda-carrageenan"	"λ-carrageenan"@en
<http://purl.obolibrary.org/obo/CHEBI_6579>	"luteolin 7-O-(6-O-malonyl-beta-D-glucoside)"	"luteolin 7-O-(6-O-malonyl-β-D-glucoside)"@en
<http://purl.obolibrary.org/obo/CHEBI_77791>	"luteolin 7-O-beta-D-glucoside(1-)"	"luteolin 7-O-β-D-glucoside(1−)"@en
<http://purl.obolibrary.org/obo/CHEBI_31799>	"malvidin 3-O-beta-D-glucoside"	"malvidin 3-O-β-D-glucoside"@en
<http://purl.obolibrary.org/obo/CHEBI_18291>	"manganese atom"	"manganese(2+)"@en
<http://purl.obolibrary.org/obo/CHEBI_25481>	"naphthoquinone"	"napthoquinone"@en
<http://purl.obolibrary.org/obo/CHEBI_28327>	"naringenin 7-O-beta-D-glucoside"	"naringenin 7-O-β-D-glucoside"@en
<http://purl.obolibrary.org/obo/CHEBI_25681>	"omega-3 fatty acid"	"ω−3 fatty acid"@en
<http://purl.obolibrary.org/obo/CHEBI_36009>	"omega-6 fatty acid"	"ω−6 fatty acid"@en
<http://purl.obolibrary.org/obo/CHEBI_25704>	"organic sulfate"	"organic sulfate salt"@en
<http://purl.obolibrary.org/obo/CHEBI_36122>	"pelargonidin 3-O-beta-D-glucoside chloride"	"pelargonidin 3-O-β-D-glucoside chloride"@en
<http://purl.obolibrary.org/obo/CHEBI_74793>	"peonidin 3-O-beta-D-glucoside"	"peonidin 3-O-β-D-glucoside"@en
<http://purl.obolibrary.org/obo/CHEBI_31985>	"petunidin 3-O-beta-D-glucoside"	"petunidin 3-O-β-D-glucoside"@en
<http://purl.obolibrary.org/obo/CHEBI_33468>	"phosphide"@en	"phosphide(3-)"
<http://purl.obolibrary.org/obo/CHEBI_26125>	"phytosterol"@en	"phytosterols"
<http://purl.obolibrary.org/obo/CHEBI_190301>	"potassium phosphate"@en	"tripotassium phosphate"
<http://purl.obolibrary.org/obo/CHEBI_8454>	"progoitrin(1-)"	"progoitrin(1−)"@en
<http://purl.obolibrary.org/obo/CHEBI_131498>	"quercetin 3,4'-di-O-beta-D-glucoside"	"quercetin 3,4'-di-O-β-D-glucoside"@en
<http://purl.obolibrary.org/obo/CHEBI_67486>	"quercetin 3-O-beta-D-galactopyranoside"	"quercetin 3-O-β-D-galactopyranoside"@en
<http://purl.obolibrary.org/obo/CHEBI_68352>	"quercetin 3-O-beta-D-glucopyranoside"	"quercetin 3-O-β-D-glucopyranoside"@en
<http://purl.obolibrary.org/obo/CHEBI_32082>	"quercetin 3-O-beta-D-glucosyl-(1->2)-beta-D-glucoside"	"quercetin 3-O-β-D-glucosyl-(1→2)-β-D-glucoside"@en
<http://purl.obolibrary.org/obo/CHEBI_75839>	"quercetin 4'-O-beta-D-glucopyranoside"	"quercetin 4ʼ-O-β-D-glucopyranoside"@en
<http://purl.obolibrary.org/obo/CHEBI_131499>	"quercetin 7,4'-di-O-beta-D-glucoside"	"quercetin 7,4'-di-O-β-D-glucoside"@en
<http://purl.obolibrary.org/obo/CHEBI_145013>	"rebaudioside B (1-)"@en	"rebaudioside B(1-)"
<http://purl.obolibrary.org/obo/CHEBI_26658>	"sesquiterpenoid"	"sesquiterpoid"@en
<http://purl.obolibrary.org/obo/CHEBI_15138>	"sulfide(2-)"	"sulfide(2−)"@en
```

**Comment by LilyAndres** [comment]
Thanks @ddooley, apologies for my late reply. I was away.

I have checked the terms and indeed some of them have a different label than what it's described in the original cheri labels. However, some of them had the same label as chebi and I'm not sure why they were highlighted as duplicated when you ran the code in FoodOn. I have fixed all of them but it would be great if you can run the command again and let me know if there are any issues. 

I'm sure we have the CHEBI terms as the ones that we want to use and we added "ExactSynonym" to describe alternative names...

<img width="1013" alt="Screen Shot 2022-11-16 at 5 50 18 pm" src="https://user-images.githubusercontent.com/25215773/202106250-1997b504-b3f4-4963-a36d-41ae9ba961e5.png">

**Comment by ddooley** [comment]
The problem is that you have AL rdfs:label@en at column top?  CHEBI uses ^^xml:string datatype instead for their labels.  That might fix duplication problem.  OBO format doesn't like for a term to have both rdfs:label@en and rdfs:label^^xml:string .

**Comment by LilyAndres** [comment]
Thanks @ddooley. I have changed the label datatype in the 'nutritional_components_framework.csv' template to AT rdfs:label^^xml:string:

<img width="999" alt="Screen Shot 2022-11-17 at 12 14 27 pm" src="https://user-images.githubusercontent.com/25215773/202329942-2009a80c-2d75-409e-b0b8-1d10bd061183.png">

Is looking something like this: 

![Screen Shot 2022-11-17 at 11 53 27 am](https://user-images.githubusercontent.com/25215773/202330144-58e0686a-0046-4d87-a2ca-830c9a8bf598.png)

I hope is ok, please let me know if not and I will make the corresponding changes. Thanks a lot.

**Comment by ddooley** [comment]
So first, I made a wee mistake - for ChEBI it should be **xsd**:string, not xml:string.  Sorry about that!

Now CDNO still has some duplicates where you've entered a new character, like for alpha-carotene?  we actually have to get to the place where every term has exactly 1 label.  From OBO principles:

```
use rdfs:label for the primary label
include exactly one rdfs:label for every declared entity (e.g. class, property)

```

So in CDNO e.g. this has to be fixed:
<img width="310" alt="image" src="https://user-images.githubusercontent.com/4000582/203409000-2e9b9054-35bf-4e27-9290-e85076da3cbb.png">

the label should match ChEBI's alpha-carotene^^xsd:string label exactly.

<rdfs:label rdf:datatype="http://www.w3.org/2001/XMLSchema#string">alpha-carotene</rdfs:label>

And the **α-carotene** label must be switched to say "alternative term" IAO_0000118 annotation, or a synonym.

**Comment by LilyAndres** [comment]
Thanks @ddooley I am a bit confused because what we used in CHEBI is the "ChEBI Name" and not what I think is the "ChEBI ASCII Name":

<img width="1077" alt="Screen Shot 2022-11-23 at 2 04 28 pm" src="https://user-images.githubusercontent.com/25215773/203462471-a4dc04d7-76f9-418d-8d33-b1525794f603.png">


@kaiiam should this be fixed in the cdno-edit.owl as well? Instead of the Robot templates?

**Comment by ddooley** [comment]
It is puzzling.  The chebi.owl and chebi-lite.owl files download directly from EMBL-EBI don't even have that α-carotene in it, they only have alpha-carotene.  
Over on Ontobee there's no hint of α-carotene , see https://ontobee.org/ontology/CHEBI?iri=http%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FCHEBI_28425  . Its as though some inaccessable version of ChEBI is driving OLS. But you guys were able to download it with that α-carotene symbol!

**Comment by LilyAndres** [comment]
Well we have created the Robot template based in the Chebi name label... I'm getting nervous about this issue. We might need to check the name of the labels again...

**Comment by LilyAndres** [comment]
@ddooley Apologies for the late reply in this issue, I just made a new release with changes that will help to fix this issue, please let me know if the issue is still there. Thanks a lot.

**Comment by LilyAndres** [comment]
We had an error while trying to compile the Nutritional_Component_Framework.csv file and I tracked it this ticket from the release "v2022-12-23". I tested all the labels by adding them again and I found that the error comes from removing the labels (second column) from the following terms in the [nutritional_component_framework.csv](https://github.com/CompositionalDietaryNutritionOntology/cdno/blob/master/src/ontology/templates/nutritional_components_framework.csv):

CHEBI:23042 "carotene"
CHEBI:28794 "coumarin"
CHEBI:28802 "flavonols"
CHEBI:24698 "hydroxyflavone"

If we delete these labels we get the following error: 

<img width="739" alt="Screen Shot 2024-05-23 at 10 08 17 pm" src="https://github.com/CompositionalDietaryNutritionOntology/cdno/assets/25215773/8f87aa12-1174-48dc-86a5-f02c1e222de2">

Any changes I try to make in the Nutritional_Component_Framework.csv file are not working until I put these labels back, run the pipeline with whatever changes I am making in this template and then it runs and makes the changes. Then I remove the labels and run the pipeline again, the changes are kept and the labels removed to avoid duplication. This is not the best thing to do but Need to work with one issue at the time.
